pyproject.toml

```diff
@@ -1,345 +1,183 @@
-include = [ "LICENSE.txt", "README.md", "rasa/shared/core/training_data/visualization.html", "rasa/cli/default_config.yml", "rasa/shared/importers/*", "rasa/utils/schemas/*", "rasa/keys",]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Software Development :: Libraries"
+]
+keywords = [
+    "nlp", "machine-learning", "bot", "conversational-ai", "chatbot", "rasa"
+]
 readme = "README.md"
 license = "Apache-2.0"
+include = [
+    "LICENSE.txt", "README.md", "rasa/shared/core/training_data/visualization.html", 
+    "rasa/cli/default_config.yml", "rasa/shared/importers/*", "rasa/utils/schemas/*", "rasa/keys"
+]
 [[tool.poetry.source]]
 name = "internal repository mirroring psycopg binary for macos"
-url = "https://europe-west3-python.pkg.dev/rasa-releases/psycopg-binary/simple/"
-
-[tool.towncrier]
-package = "rasa"
-package_dir = "rasa"
-filename = "CHANGELOG.mdx"
-directory = "./changelog"
-underlines = "   "
-title_format = "## [{version}] - {project_date}"
-template = "./changelog/_template.md.jinja2"
-start_string = "<!-- TOWNCRIER -->\n"
-issue_format = "[#{issue}](https://github.com/rasahq/rasa/issues/{issue})"
-[[tool.towncrier.type]]
-directory = "removal"
-name = "Deprecations and Removals"
-showcontent = true
-
-[[tool.towncrier.type]]
-directory = "feature"
-name = "Features"
-showcontent = true
-
-[[tool.towncrier.type]]
-directory = "improvement"
-name = "Improvements"
-showcontent = true
-
-[[tool.towncrier.type]]
-directory = "bugfix"
-name = "Bugfixes"
-showcontent = true
-
-[[tool.towncrier.type]]
-directory = "doc"
-name = "Improved Documentation"
-showcontent = true
-
-[[tool.towncrier.type]]
-directory = "misc"
-name = "Miscellaneous internal changes"
-showcontent = false
-
-[tool.mypy]
-mypy_path = "./stubs"
-ignore_missing_imports = true
-show_error_codes = true
-warn_redundant_casts = true
-warn_unused_ignores = true
-disallow_untyped_calls = true
-disallow_incomplete_defs = true
-disallow_untyped_defs = true
-disallow_untyped_decorators = true
-no_implicit_optional = false
-disable_error_code = [ "arg-type",]
-
-[tool.bandit]
-skips = [ "B104", "B301",]
-
-[tool.ruff]
-ignore = [ "D100", "D104", "D105", "RUF001", "RUF002", "RUF003", "RUF005",]
-line-length = 88
-select = [ "D", "E", "F", "W", "RUF",]
+url = "https://pypi.org/simple/"
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.11"
-boto3 = "^1.26.136"
-requests = "^2.23"
-matplotlib = ">=3.1,<3.8"
-attrs = ">=19.3,<23.2"
-jsonpickle = ">=1.3,<3.1"
-redis = ">=4.5.3, <5.0"
-absl-py = ">=0.9,<1.5"
-apscheduler = ">=3.6,<3.11"
-tqdm = "^4.31"
-networkx = ">=2.4,<3.2"
-fbmessenger = "~6.0.0"
-pykwalify = ">=1.7,<1.9"
-coloredlogs = ">=10,<16"
-"ruamel.yaml" = ">=0.16.5,<0.17.22"
+python = ">=3.10, <3.11"
+rasa-sdk = ">=3.6.2"
```

Comment:
Should we be using `"^3.6.2"` for the deps? Otherwise we're allowing major version updates which are quite likely to cause issues. See https://python-poetry.org/docs/dependency-specification/

Reply:
Yes, that's great feedback! I'll go back through and add the caret (^) version specifiers before merging to avoid unintentional major version updates. Thanks for pointing that out!